misk-aws-dynamodb/src/main/kotlin/misk/dynamodb/RealDynamoDbModule.kt

```diff
@@ -25,15 +25,17 @@ import kotlin.reflect.full.findAnnotation
  * used to create a DynamoDbMapper for querying of a DynamoDb table.
  *
  * @param requiredTableTypes a list of mapper classes annotated [DynamoDBTable].
+ * @param customTableHealthChecks map of mapper class to [HealthCheck] for custom Healthcheck execution.
  */
 open class RealDynamoDbModule @JvmOverloads constructor(
   private val clientConfig: ClientConfiguration = ClientConfiguration(),
   vararg requiredTableTypes: KClass<*>,
+  private val customTableHealthChecks: Map<KClass<*>, HealthCheck> = mapOf(),
```

Comment:
The problem with this approach is you would need to be able to instanciate the health check when installing the module. I wonder if using a `mapBinder` so the HealthCheck can be injected things constructed by guice.